src/state.rs

```diff
@@ -0,0 +1,22 @@
+/// The state of a single unit
+#[derive(RustcEncodable, RustcDecodable, PartialEq)]
+pub struct Unit {
+    pub position: [u64; 2],
+    pub angle: u64,
+    pub speed_vector: [u64; 2],
```

Comment:
Yeah, that looks good :) Didn't think about the case of a unit standing still.

Btw, the angle should probably be in radians, so f64.